### PR TITLE
Strip rollup-plugin prefix to find named plugin exports, throw when export cannot be found

### LINKED ISSUE #3646: Rollup does not error when full plugin name is passed onto the command line
- Rollup Version: 2.18.0
- Operating System (or Browser): n/a
- Node Version (if applicable): n/a
- Link to reproduction (IMPORTANT, read below): https://repl.it/repls/DistortedBasicJavadocs#.replit

### Expected Behavior

Specifying a plugin name that does not have any effect should return an exit code 1. E.g. The following command should fail:

```bash
npx rollup -c --plugin rollup-plugin-terser
```

For the code snippet:

```js
console.log('main entry');

const this_should_be_minified = 42;

function addAFunction() {
  return this_should_be_minified;
}

console.log(addAFunction());
```

### Actual Behavior

The commandline silently accepts the name, but does not do anything. E.g. it rolls up, but does not minify and keeps the original source file intact.

The correct version should have been

```bash
npx rollup -c --plugin terser
```

If you run this command, then it correctly minifies and exits cleanly:

```js
console.log("main entry");console.log(42);
```

### PR DESCRIPTION
**Pull request by lukastaegert**
This PR contains:
- [x] bugfix
- [ ] feature
- [ ] refactor
- [ ] documentation
- [ ] other

Are tests included?
- [x] yes (*bugfixes and features will not be merged without tests*)
- [ ] no

Breaking Changes?
- [ ] yes (*breaking changes will not be merged unless absolutely necessary*)
- [x] no

List any relevant issue numbers:
Resolves #3646 

### Description
This will improve how Rollup handles exports of plugins passed via the `--plugin` option to detect a potential named export (if no "default" export is available):
- If there is a `rollup-plugin-` or `@rollup/plugin-` prefix it will be stripped
- If there is a path, it will only take the part after the last `\` or `/`
- If there is a `.`, it will only take the part before the first `.`
- The remaining part will be converted from kebab (`my-plugin-name`) to camel case (`myPluginName`)

E.g.
- `rollup-plugin-terser`: Check `terser`
- `@rollup/plugin-node-resolve`: Check `nodeResolve`
- `./path/to/my-awesome-plugin`: Check `myAwesomePlugin`
- `./path/to/my-awesome-plugin.js`: Check `myAwesomePlugin`

Furthermore if no sensible export can be found, it now throws an error indicating exactly which exports were tried.

### CODE REVIEW

File: cli/run/commandPlugins.ts [hunk 1]
```diff
@@ -63,16 +60,34 @@ function loadAndRegisterPlugin(inputOptions: InputOptions, pluginText: string) {
...
+		);
 	}
 	inputOptions.plugins!.push(
 		typeof plugin === 'function' ? plugin.call(plugin, pluginArg) : plugin
 	);
 }
+
+function getCamelizedPluginBaseName(pluginText: string): string {
+	return pluginText
+		.match(/^(@rollup\/plugin-|rollup-plugin-)?(.*)$/)![2]
```
Comment (kzc) [inline cli/run/commandPlugins.ts:86]: If the `^` is removed from the regex then paths such as the following could be correctly handled:
```
./node_modules/@rollup/plugin-node-resolve
/absolute/dir/rollup-plugin-node-resolve
```
Also, replacing `(.*)` with `([-\w]+)` would be better - has at least one non-dash, non-space character.

Comment (shellscape) [reply cli/run/commandPlugins.ts:86]: I'd rather leave filesystem resolution up to node. if we assert a package name, users could alias that package to their file system using package.json

Comment (kzc) [reply cli/run/commandPlugins.ts:86]: Ignore the `node_modules` and `node-resolve` in the examples - they could be anything. My comment is unrelated to node file resolution. It solely a `--plugin` parameter parsing issue. The CLI plugin parameter already supports relative and absolute path arguments. Look at the tests for examples.

Comment (lukastaegert) [reply cli/run/commandPlugins.ts:86]: Makes sense. I will add some changes.

Comment (kzc) [reply cli/run/commandPlugins.ts]: lgtm

**Pull request merged**